### PR TITLE
Add parallel netcdf capability for regional grids

### PR DESCRIPTION
**Pull request by junwang-noaa**
## Description

1. add parallel netcdf call for regional fv3 grids
2. Change restart_interval to real from integer so that model can output restart files at non-integer forecast time, e.g. output restart files after the first time step.
3. Set forecast time to be real to allow flexible forecast time.

### Issue(s) addressed

[FV3 issue #174](https://github.com/NOAA-EMC/fv3atm/issues/174)

## Testing

The code will be tested for LAM and HAFS high resolution runs.
RT will be conducted on supported platforms, the current baseline results will not change.

## Dependencies

[fv3 PR #175](https://github.com/NOAA-EMC/fv3atm/pull/175)
[NEMS PR #81](https://github.com/NOAA-EMC/NEMS/pull/81)
[ufs-weather-model PR#206](https://github.com/ufs-community/ufs-weather-model/pull/206)

### CODE REVIEW

**Comment by junwang-noaa** [comment]
Bin, I saw you only ran RT on hera and orion in yoru PR #256. But the datm tests were added to dell too, my RT failed on dell  with datm tests because there is no datm baseline. I am going to comment out these tests on dell in my commit #206. 

++ cp /gpfs/dell2/emc/modeling/noscrub/emc.nemspara/RT/NEMSfv3gfs/develop-20201022/DATM/cfsr.SCRIP.nc DATM_INPUT
cp: cannot stat '/gpfs/dell2/emc/modeling/noscrub/emc.nemspara/RT/NEMSfv3gfs/develop-20201022/DATM/cfsr.SCRIP.nc': No such file or directory

**Comment by binli2337** [comment]
@junwang-noaa The RTs have not been run on wcoss_dell_p3 before. I have transferred input files for DATM RTs on wcoss_dell_p3. New baseline will be available in the next PR for DATM tests.

**Review by binli2337 — APPROVED** [review]
Looks good. Thanks.

**Review by DusanJovic-NOAA — APPROVED** [review]

File: tests/rt.conf [hunk 1]
```diff
@@ -61,6 +61,7 @@ COMPILE | CCPP=Y SUITES=FV3_GFS_2017_gfdlmp_regional,FV3_GFS_2017_gfdlmp_regiona
 RUN     | fv3_ccpp_regional_control                                                                                                      | standard    |                | fv3         |
 RUN     | fv3_ccpp_regional_restart                                                                                                      | standard    |                | fv3         | fv3_ccpp_regional_control
 RUN     | fv3_ccpp_regional_quilt                                                                                                        | standard    |                | fv3         |
+#RUN     | fv3_ccpp_regional_quilt_netcdf_parallel                                                                                        | standard    |                | fv3         |
```
Comment (BinLiu-NOAA) [inline tests/rt.conf:64]: @junwang-noaa @DusanJovic-NOAA Does this mean that by default the fv3_ccpp_regional_quilt_netcdf_parallel test will not be included in the regression tests?

**Comment by junwang-noaa** [comment]
Bin, this test needs to use nccmp for file comparison, the capability is in
Minsuk's PR#261. I just added a comment in his PR, so that we can turn on
the two parallel netcdf tests there.

**Review by BinLiu-NOAA — APPROVED** [review]
Looks good to me.

**Pull request merged**